adapters/audienceNetwork/facebook.go

```diff
@@ -37,267 +41,405 @@ func (a *FacebookAdapter) Name() string {
-		return
+
+	if request.User == nil || request.User.BuyerUID == "" {
+		return nil, []error{&errortypes.BadInput{
+			Message: "Missing bidder token in 'user.buyeruid'",
+		}}
 	}
 
-	result.StatusCode = anResp.StatusCode
+	return this.buildRequests(request)
+}
 
-	defer anResp.Body.Close()
-	body, _ := ioutil.ReadAll(anResp.Body)
-	result.ResponseBody = string(body)
+func (this *FacebookAdapter) buildRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	// Documentation suggests bid request splitting by impression so that each
+	// request only represents a single impression
+	reqs := make([]*adapters.RequestData, 0, len(request.Imp))
+	headers := http.Header{}
+	var errs []error
+
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("Auth-Token", this.appID+"|"+this.appSecret)
+	headers.Add("X-Fb-Pool-Routing-Token", request.User.BuyerUID)
+
+	for _, imp := range request.Imp {
+		// Make a copy of the request so that we don't change the original request which
+		// is shared across multiple threads
+		fbreq := *request
+		fbreq.Imp = []openrtb.Imp{imp}
+
+		if err := this.modifyRequest(&fbreq); err != nil {
+			errs = append(errs, err)
+			continue
+		}
 
-	if anResp.StatusCode == http.StatusBadRequest {
-		err = &errortypes.BadInput{
-			Message: fmt.Sprintf("HTTP status %d; body: %s", anResp.StatusCode, result.ResponseBody),
+		body, err := json.Marshal(&fbreq)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		body, err = modifyImpCustom(body, &fbreq.Imp[0])
+		if err != nil {
+			errs = append(errs, err)
+			continue
 		}
-		return
-	}
 
-	if anResp.StatusCode == http.StatusNoContent {
-		return
+		reqs = append(reqs, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     this.URI,
+			Body:    body,
+			Headers: headers,
+		})
 	}
 
-	if anResp.StatusCode != http.StatusOK {
-		err = &errortypes.BadServerResponse{
-			Message: fmt.Sprintf("HTTP status %d; body: %s", anResp.StatusCode, result.ResponseBody),
-		}
-		return
+	return reqs, errs
+}
+
+// The authentication ID is a sha256 hmac hash encoded as a hex string, based on
+// the app secret and the ID of the bid request
+func (this *FacebookAdapter) makeAuthID(req *openrtb.BidRequest) string {
+	h := hmac.New(sha256.New, []byte(this.appSecret))
+	h.Write([]byte(req.ID))
+
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func (this *FacebookAdapter) modifyRequest(out *openrtb.BidRequest) error {
+	if len(out.Imp) != 1 {
+		panic("each bid request to facebook should only have a single impression")
 	}
 
-	var bidResp openrtb.BidResponse
-	err = json.Unmarshal(body, &bidResp)
+	imp := &out.Imp[0]
+	plmtId, pubId, err := this.extractPlacementAndPublisher(imp)
 	if err != nil {
-		err = &errortypes.BadServerResponse{
-			Message: err.Error(),
-		}
-		return
+		return err
 	}
-	if len(bidResp.SeatBid) == 0 {
-		return
+
+	reqExt := facebookReqExt{
+		PlatformID: this.platformID,
```

Comment:
I couldn't find where this platformID is defined, think this can be defined inside server since it will be identical across your bid requests.